sky/clouds/kubernetes.py

```diff
@@ -413,12 +415,21 @@ def make_deploy_resources_variables(
         # Larger timeout may be required for autoscaling clusters, since
         # autoscaler may take some time to provision new nodes.
         # Note that this timeout includes time taken by the Kubernetes scheduler
-        # itself, which can be upto 2-3 seconds.
-        # For non-autoscaling clusters, we conservatively set this to 10s.
+        # itself, which can be upto 2-3 seconds, and up to 10-15 seconds when
+        # scheduling 100s of pods.
+        # For non-autoscaling clusters, we conservatively set this to 30s.
         timeout = skypilot_config.get_nested(
             ('kubernetes', 'provision_timeout'),
-            10,
+            30,
```

Comment:
Can we make this provision_timeout to scale based on the number of nodes we request? This can make the failover faster for high-end GPUs